internal/artifacts/module_arch.go

```diff
@@ -22,35 +22,73 @@ const (
 
 // ExecuteBuild - executes build of module from Makefile
 func ExecuteBuild(source, target string, extensions []string, moduleName, platform string, wdGetter func() (string, error)) error {
+	if moduleName == "" {
+		return errors.New(buildFailedOnEmptyModuleMsg)
+	}
+
 	logs.Logger.Infof(buildMsg, moduleName)
 	loc, err := dir.Location(source, target, dir.Dev, extensions, wdGetter)
 	if err != nil {
 		return errors.Wrapf(err, buildFailedMsg, moduleName)
 	}
-	err = buildModule(loc, loc, moduleName, platform)
+	err = buildModule(loc, loc, moduleName, platform, true)
 	if err != nil {
 		return err
 	}
 	logs.Logger.Infof(buildFinishedMsg, moduleName)
 	return nil
 }
 
-// ExecuteSoloBuild - executes build of module from stand along command
+// ExecuteSoloBuild - executes build of module from stand alone command
 func ExecuteSoloBuild(source, target string, extensions []string, moduleName string, wdGetter func() (string, error)) error {
 	logs.Logger.Infof(buildMsg, moduleName)
-	loc, err := dir.Location(source, target, dir.Dev, extensions, wdGetter)
+
+	sourceDir, err := getSoloModuleBuildAbsSource(source, wdGetter)
 	if err != nil {
 		return errors.Wrapf(err, buildFailedMsg, moduleName)
 	}
-	targetLoc := dir.ModuleLocation(loc, target)
-	err = buildModule(loc, targetLoc, moduleName, "cf")
+
+	targetDir, err := getSoloModuleBuildAbsTarget(sourceDir, target, moduleName, wdGetter)
+	if err != nil {
+		return errors.Wrapf(err, buildFailedMsg, moduleName)
+	}
+
+	loc, err := dir.Location(sourceDir, targetDir, dir.Dev, extensions, wdGetter)
+	if err != nil {
+		return errors.Wrapf(err, buildFailedMsg, moduleName)
+	}
+	targetLoc := dir.ModuleLocation(loc)
+	err = buildModule(loc, targetLoc, moduleName, "cf", false)
```

Comment:
I would send an empty string instead of `cf` here (it isn't validated anyway and if for some reason that changes, we'll see it as an error of a failed test instead of a zip file which isn't packaged to the mtar, which is harder to find the reason to quickly)

Reply:
done